pyannote/database/custom.py

```diff
@@ -169,17 +169,25 @@ def subset_iter(database_name, file_lst=None, file_rttm=None,
         # initialize current file with the mandatory keys
         current_file = {'database': database_name, 'uri': uri}
 
+        # add 'annotated' when UEM file is provided
+        # defaults to empty Timeline for the same reason as above
+        if file_uem is not None:
+            current_file['annotated'] = annotated.get(uri, Timeline(uri=uri))
+
         # add 'annotation' when RTTM file is provided
         # defaults to empty Annotation because of
         # github.com/pyannote/pyannote-database/pull/13#discussion_r261564520)
         if file_rttm is not None:
             current_file['annotation'] = annotations.get(
                 uri, Annotation(uri=uri))
-
-        # add 'annotated' when UEM file is provided
-        # defaults to empty Timeline for the same reason as above
-        if file_uem is not None:
-            current_file['annotated'] = annotated.get(uri, Timeline(uri=uri))
+            if current_file['annotated']:
+                annotation = current_file['annotation'].crop(current_file['annotated'])
```

Comment:
I'd rather check if a crop is needed before doing it.

I thought there was an easy way to check if a timeline covers another timeline but apparently there is none.
We probably should add such a method to `pyannote.core.Timeline` and then use something like:
```python
if not annotated.covers(annotation.get_timeline()):
    # do the crop
```

Reply:
`self.covers(other)` could be implemented by checking that `self.gaps(support=other.extent()).co_iter(other)` does not yield anything. This has to be tested though...

Reply:
In the  current version, `crop` will be called even if it is not needed (i.e. `annotation` is fully included in `annotated`) and will therefore slow the whole thing down for nothing in most use cases.

One solution is to first check whether `crop` is needed and, only do the cropping if it is.

This is only a good solution if checking is faster than cropping.
Would you mind benchmarking the proposed solution above? (the one that involves `covers`).

Reply:
right, I tested with the first episode of friends, using two different protocols (one where the annotation is fully included in the UEM and one where it is not)

protocol/method| covers | crop
--|--|--
fully included | 1.48 ms | 4.73 ms
not fully included | 5.68 ms | 33.6 ms

I used `%%timeit` to get the time
```py
%%timeit -n 1000
annotated.covers(reference.get_timeline())

%%timeit -n 100
reference.crop(annotated)
```

Reply:
Thanks! Can you please share your implementation of `Timeline.covers` ?